server/src/main/java/keywhiz/service/daos/SecretDAO.java

```diff
@@ -170,6 +169,7 @@ public long createSecret(String name, String encryptedSecret,
   public ImmutableList<SimpleEntry<Long, String>> getSecretsNameOnly() {
     List<SimpleEntry<Long, String>> results = dslContext.select(SECRETS.ID, SECRETS.NAME)
         .from(SECRETS)
+        .where(SECRETS.CURRENT.isNotNull())
```

Comment:
It's currently the case for all secrets (except the few broken ones). Going forward, we will stop deleting secrets and set current to null instead.